core/src/main/c/share/vec_agg.cpp

```diff
@@ -167,29 +167,34 @@ int64_t SUM_LONG(int64_t *pl, int64_t count) {
 int64_t MIN_LONG(int64_t *pl, int64_t count) {
     Vec8q vec;
     const int step = 8;
-    Vec8q vecMin = L_MAX;
+    Vec8q vecMin = L_MIN;
     Vec8qb bVec;
     int i;
     for (i = 0; i < count - 7; i += step) {
         _mm_prefetch(pl + i + 63 * step, _MM_HINT_T1);
         vec.load(pl + i);
-        bVec = vec == L_MIN;
-        vecMin = select(bVec, vecMin, min(vecMin, vec));
+        vecMin = select(vecMin == L_MIN, vec, vecMin);
+        vec = select(vec == L_MIN, vecMin, vec);
+        vecMin = min(vec, vecMin);
```

Comment:
How would this calculate min when initial value is the lowest possible?

Reply:
Both previous `select()`s make sure that we may only see `(vec == L_MIN && vecMin == L_MIN) == true` or `(vec != L_MIN && vecMin != L_MIN) == true` when we come to this line